src/string_bytes.cc

```diff
@@ -72,10 +73,15 @@ class ExternString: public ResourceType {
       ExternString* h_str = new ExternString<ResourceType, TypeName>(isolate,
                                                                      data,
                                                                      length);
-      Local<String> str = String::NewExternal(isolate, h_str);
+      MaybeLocal<String> str = String::NewExternal(isolate, h_str);
       isolate->AdjustAmountOfExternalAllocatedMemory(h_str->byte_length());
 
-      return scope.Escape(str);
+      if (str.IsEmpty()) {
+        delete h_str;
+        return Local<String>();
+      }
```

Comment:
Only `New` calls `AdjustAmountOfExternalAllocatedMemory` and I think it's not called in the constructor of `ExternString` because it makes only sense after `String::NewExternal`.